### PR TITLE
feat(OTP): Updated the API to use the graphql endpoint of OTP

### PR DESCRIPTION
**Pull request by kotva006**
#### Summary of changes
This pull request changes how we interact with OTP
I moves from the deprecated rest API, to the GraphQL API

This is done by adding a new client library `absinthe_client` and updating its dependencies.
Most of the fields from the REST API response match the fields in the GraphQL shape.  Some had to be updated
Some query fields also changed name and format.  Most notably the `transportationModes`
So unsupported and unneeded fields got removed as well.

**Asana Ticket:** [Switch dotcom over from using OpenTripPlanner's REST API to the GQL API](https://app.asana.com/0/home/1201123880594717/1204759745777344)

---

#### General checks
* [ ] **Are the changes organized into self-contained commits with descriptive and well-formatted commit messages?** This is a good practice that can facilitate easier reviews.
* [ ] **Testing.** Do the changes include relevant passing updates to tests? This includes updating screenshots. Preferably tests are run locally to verify that there are no test failures created by these changes, before opening a PR.
* [ ] **Tech debt.** Have you checked for tech debt you can address in the area you're working in? This can be a good time to address small issues, or create Asana tickets for larger issues.

#### New UI, or substantial UI changes
* [ ] **Cross-browser compatibility** is less of an issue now that we're no longer supporting IE, but changes still need to work as expected in Safari, Chrome, and Firefox.
* [ ] **Are interactive elements accessible?** This includes at minimum having [relevant keyboard interactions](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#keyboard-interaction) and [visible focus](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#visible-focus), but can also include verification with screen reader testing.
* [ ] **Other accessibility checks** such as sufficient [color constrast](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#color-contrast), or whether the layout holds up at 200% [zoom level](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#visible-focus).

#### New endpoints, or non-trivial changes to current endpoints
* [ ] **Have we load-tested** any new pages or internal API endpoints that will receive significant traffic? [See load testing docs](./../apps/site/load_tests/README.md)
* [ ] **If this change involves routes,** does it work correctly with pertinent "unusual" routes such as the combined Green Line, Silver Line, Foxboro commuter rail, and single-direction bus routes like the 170?

### CODE REVIEW

File: apps/trip_plan/lib/trip_plan/api/open_trip_planner/builder.ex [hunk 1]
```diff
@@ -1,26 +1,40 @@
...
+  def build_query_params(from, to, opts) do
+    from_string = location(from)
+    to_string = location(to)
+    default_mode_string = "[{mode: WALK}, {mode: TRANSIT}]"
+
+    do_build_params(opts, %{
+      "fromPlace" => from_string,
+      "toPlace" => to_string,
+      "transportModes" => default_mode_string,
+      "walkReluctance" => 5,
```
Comment (thecristen) [inline apps/trip_plan/lib/trip_plan/api/open_trip_planner/builder.ex:18]: **suggestion**: The results are super different with no walk distance configured. After playing around with various itineraries it looks like bumping up this value can compensate for that. I think `15` produced a nice result.

File: apps/trip_plan/lib/trip_plan/api/open_trip_planner/builder.ex [hunk 2]
```diff
@@ -1,26 +1,40 @@
...
+
   @doc "Convert general planning options into query params for OTP"
   @spec build_params(Position.t(), Position.t(), TripPlan.Api.plan_opts()) ::
           {:ok, %{String.t() => String.t()}} | {:error, any}
   def build_params(from, to, opts) do
-    opts = configure_walk_distance(from, to, opts)
     do_build_params(opts, %{"mode" => "TRANSIT,WALK", "walkReluctance" => 5})
   end
 
-  defp configure_walk_distance(from, to, opts) do
```
Comment (thecristen) [inline apps/trip_plan/lib/trip_plan/api/open_trip_planner/builder.ex:14]: FYI there are some unused variables to clean up after removing this.

File: apps/trip_plan/lib/trip_plan/api/open_trip_planner/builder.ex (hunk 1 above)
Comment (thecristen) [reply apps/trip_plan/lib/trip_plan/api/open_trip_planner/builder.ex:18]: though if you increase that you'll want to increase the value in line 74 (for `{:optimize_for, :less_walking}`) to be even greater!

I also recognize these are totally subjective things, but in this case I was aiming for comparable results to current state.

**Review by thecristen — COMMENTED** [review]
This is awesome! Here's some initial comments. See also #1792

File: apps/trip_plan/lib/trip_plan/api/open_trip_planner.ex [hunk 3]
```diff
@@ -86,55 +84,133 @@ defmodule TripPlan.Api.OpenTripPlanner do
...
-        "#{__MODULE__}.plan_response url=#{url} is_otp2=#{String.contains?(url, config(:otp2_url))} params=#{inspect(params)} #{status_text(response)} duration=#{duration / :timer.seconds(1)}"
+        "#{__MODULE__}.plan_response url=#{url} is_otp2=#{String.contains?(url, config(:otp2_url))} query=#{inspect(query)} #{status_text(response)} duration=#{duration / :timer.seconds(1)}"
       end)
 
     response
   end
 
-  defp build_headers("true") do
-    proxy_url = Application.get_env(:trip_plan, OpenTripPlanner)[:wiremock_proxy_url]
-    [{"X-WM-Proxy-Url", proxy_url}]
```
Comment (kotva006) [inline apps/trip_plan/lib/trip_plan/api/open_trip_planner.ex:120]: @thecristen: Do we still need this header?

Comment (thecristen) [reply apps/trip_plan/lib/trip_plan/api/open_trip_planner.ex:120]: Let's keep it; I'm not ready to give up on Wiremock (Java app that can record and save API responses for testing) yet despite the fact we don't rely on it at the moment...

**Review by thecristen — COMMENTED** [review]
Awesome! Three more things, the latter two which are small chores:

| Dev (current state) | Dev-green (this PR) |
| --- | --- |
| image  | image  | 

1. I noticed the "no trips found state" looks different. The example above is using [these inputs](https://dev-green.mbtace.com/trip-planner?plan%5Bfrom%5D=Fall+River+Historical+Society%2C+451+Rock+St%2C+Fall+River%2C+MA%2C+02720%2C+USA&plan%5Bfrom_latitude%5D=41.70774&plan%5Bfrom_longitude%5D=-71.1513&plan%5Bto%5D=Middleborough%2C+MA%2C+USA&plan%5Bto_latitude%5D=41.89443&plan%5Bto_longitude%5D=-70.90686#plan_result_focus) from `Fall River Historical Society, 451 Rock St, Fall River, MA, 02720, USA` to `Middleborough, MA, USA`. Preferably it would be unchanged. Do you have a sense to what's causing the discrepancy, and is it fixable? 
2. Several Elixir unit tests need to be updated with the new `walkRelutance` values.
3. Another test is failing with a `nil` stop ID
```elixir
  8) test parse_json/1 an itinerary has legs (TripPlan.Api.OpenTripPlanner.ParserTest)
     apps/trip_plan/test/api/open_trip_planner/parser_test.exs:38
     match (=) failed
     code:  assert %NamedPosition{stop_id: "70016"} = walk_leg.from
     left:  %TripPlan.NamedPosition{stop_id: "70016"}
     right: %TripPlan.NamedPosition{
              stop_id: nil,
              latitude: 42.349662,
              longitude: -71.063917,
              name: "Tufts Medical Center - Outbound"
            }
     stacktrace:
       test/api/open_trip_planner/parser_test.exs:53: (test)

```

**Review by thecristen — APPROVED** [review]
🎉 sweet!
